ios/Classes/GoogleMapsNavigationView.swift

```diff
@@ -48,24 +48,64 @@ class GoogleMapsNavigationView: NSObject, FlutterPlatformView, ViewSettledDelega
   private var _consumeMyLocationButtonClickEventsEnabled: Bool = false
   private var _listenCameraChanges = false
   var isAttachedToSession: Bool = false
+  private let _isCarPlayView: Bool
 
   func view() -> UIView {
     _mapView
   }
 
+  // Getter that wont return viewEventApi if viewId is missing.
+  private func getViewEventApi() -> ViewEventApi? {
+    if _viewId != nil {
+      return _viewEventApi
+    }
+    return nil
+  }
+
+  // CarPlay initializer. This will ignore viewId and viewEventApi.
+  init(frame: CGRect,
```

Comment:
This is mostly duplicate with the next method, causing potential future issues as same code need be written twice.
Why not change the init so that viewId and viewEventApi can be null.
And maybe even have isCarPlay flag in the init params.

Reply:
I initially had just one initializer but I changed to this as I was worried that with one initializer we might end up in some weird state where either viewId or viewEvent Api is defined but not the other one. Meaning state where we are not clearly in either carplay nor regular map view state.

If you think it's better to have one initializer and separate things with isCarPlay flag, I can change it.

Reply:
Done.